Chess: Fix stockfish interaction, set I/O as non-blocking on demand

BenWiederhake opened the pull request:
This fixes the behavior of Chess, ChessEngine, and stockfish, as discovered in #18946.

HOWEVER: This causes some other funky issues. 
- Running ChessEngine in the Terminal causes weird behavior, because apparently ChessEngine setting its fd to non-blocking causes some zero-reads in either Shell or Terminal, which makes the Terminal terminate on the second input line. Huh.
- This raises the question why `BufferedFile::can_read_line` insists on populating its entire buffer in the first place. Perhaps that should be fixed, and the `Chess` pipes should always and everywhere be nonblocking?

BenWiederhake commented:
Changes since last push:
- Rebased for convenience, and to clear out CI flakes (or rather, fallout from #18887)

@tcl3, can you confirm that this makes Chess+stockfish work again?

tcl3 commented:
I can confirm that this change makes `stockfish` work as expected :tada::tada::tada: 

In addition to running `ChesEngine` using `Shell` in the Terminal  (which causes the Terminal to close), I also tried:
* Running it in the SerenityOS Terminal with the `bash` port. `bash` will exit after the first command (without closing the terminal) 
* Including `ChessEngine` in the lagom build and running it in the Linux terminal. `ChessEngine` works as expected here.

This would suggest there is an issue with the Terminal handling non-blocking fds, as you say.

ADKaster commented:
Do we want to mimimize and document the mentioned terminal issue? It'll probably end up being in either LibLine or LibVT.

BenWiederhake commented:
Changes since last push:
- Fixed a weird line wrap – Thanks @ADKaster!
- Documented that `set_blocking()` shouldn't be used in Lagom. Thanks @ADKaster!
- Rebased for convenience

I don't really think that minimizing has much benefit. Whoever wants to reproduce the issue can just run `ChessEngine`, enter `uci`, and observe the early termination. I can create an issue if you like though.
